### DISCUSSION TITLE
adafruit-nrfutil

### DISCUSSION THREAD

**Discussion by sethmnone** [markqvist/Reticulum#548] [Help & Questions]
![image](https://github.com/user-attachments/assets/5ed6f0d7-9739-49fc-8411-2505f1e3a511)

Hi all!

I seem to have hit a barrier... I have installed adafruit-nrfutil.
What am I missing?

**Comment by markqvist** [comment]
Pinging @jacobeva, since he might know whats going on here.

**Reply by sethmnone** [reply]
Thanks for checking in! Would it be a better course, for me to just sideload LINUX and continue working in python on LINUX?
What OS are you running?

**Comment by jacobeva** [comment]
That's weird. Never tested it on Windows unfortunately. OP, does running adafruit-nrfutil work in your powershell instance? If not, you probably need to set your system path to include the binary for adafruit-nrfutil somehow. Been a while since I've used python on Windows.

**Reply by sethmnone** [reply]
I can check that today, but I will pose the same question: should I just sideload LINUX and run python from there?
If windows is the issue I believe it can be easily solved. What OS are you using for rns+rnconf?

**Comment by jacobeva** [comment]
Yes, Linux definitely works. That's what I'm currently using :)

**Reply by sethmnone** [reply]
That's probably the easiest solution TBH.

**Reply by sethmnone** [reply]
> Yes, Linux definitely works. That's what I'm currently using :)
> […](#)
> On 24/09/12 08:56am, Seth C wrote: I can check that today, but I will pose the same question: should I just sideload LINUX and run python from there? If windows is the issue I believe it can be easily solved. What OS are you using for rns+rnconf? -- Reply to this email directly or view it on GitHub: [#548 (reply in thread)](https://github.com/markqvist/Reticulum/discussions/548#discussioncomment-10628110) You are receiving this because you were mentioned. Message ID: ***@***.***>

Ok, Linux Mint installed... first warning, "externally managed environment". Is it normal to install packages system-wide?

I ended up w/ pipx install but ran into permissions errors when trying to open the serial port. I ran through a number of permission access commands but I ended up hitting a barrier with adafruit-nrfutil again. Seems like another path issue.

So back to my 1st question, should I just use pip's argument --break-system-packages ....
and go from there?

Thx!

**Reply by faragher** [reply]
I use --break-system-packages. I've had zero luck with pipx.

If you can't open the serial port, you may need to add your user to the `dialout` group and log out/in.

`sudo usermod -a -G dialout <user>`

**Reply by sethmnone** [reply]
Hey! Thanks for the reply. I did add my user to the dialout group... but as mentioned, in pipx the return I receive is that I am still missing adafruit. I know it's a path issue now.

Being that you use --break-system that helps me quite a bit.

Thanks for your input about pipx! I will use pip tonite and report back any issues I have.

**Comment by jacobeva** [comment]
Yes on both accounts.

**Reply by sethmnone** [reply]
Thanks again, I think I will be g2g tonite when I install via pip / --break-system-packages.

I appreciate the help and quick response from the Dev's! @markqvist thanks again!
